### PR TITLE
Support import gzip json file by command

### PR DESCRIPTION
**Pull request by ndinh215**
I created new PR. Please review this implemention. Thanks.

### CODE REVIEW

File: Command/IndexImportCommand.php [hunk 1]
```diff
@@ -53,13 +59,19 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+        }
+
         /** @var ImportService $importService */
         $importService = $this->getContainer()->get('es.import');
         $importService->importIndex(
             $manager,
             $input->getArgument('filename'),
             $output,
-            $input->getOption('bulk-size')
+            $input->getOption('bulk-size'),
```
Comment (murnieza) [inline Command/IndexImportCommand.php]: `bulk-size` could also be in `$options`.

**Comment by ndinh215** [comment]
closing #382

**Comment by murnieza** [comment]
That's just a thought. Maybe we should leave it as a separated argument as it is required. Opinions?

File: Command/IndexImportCommand.php [hunk 2]
```diff
@@ -53,13 +59,19 @@ protected function execute(InputInterface $input, OutputInterface $output)
     {
         $manager = $this->getManager($input->getOption('manager'));
 
+        $options = [];
+        if ($input->getOption('gzip')) {
+            $options['gzip'] = true;
```
Comment (mvar) [inline Command/IndexImportCommand.php]: You should set it to `null` with current implementation. With `true`/`false` it behaves the same, as with any other value.

**Comment by ndinh215** [comment]
@saimaz  & @mvar , how do you think? About me, its rather appropriate.

File: Command/IndexImportCommand.php (hunk 2 above)
Comment (ndinh215) [inline Command/IndexImportCommand.php]: Yup, regardless of the option's value, I only check the existence of the key 'gzip'. Your idea is nice. I will fix it. Thanks.

File: Command/IndexImportCommand.php (hunk 1 above)
Comment (saimaz) [inline Command/IndexImportCommand.php]: Actually, it must be in `$options`, that's why this variable was introduced, to collect all options in a single place.

File: Service/Json/JsonReader.php [hunk 3]
```diff
@@ -66,18 +66,26 @@ class JsonReader implements \Countable, \Iterator
...
      * Constructor.
      *
      * @param Manager $manager
      * @param string  $filename
+     * @param array   $options
      * @param bool    $convertDocuments
+     *
      */
-    public function __construct($manager, $filename, $convertDocuments = true)
+    public function __construct($manager, $filename, $options, $convertDocuments = true)
```
Comment (saimaz) [inline Service/Json/JsonReader.php]: $convertDocuments might be good to put in `$options` too. Ideas? @ndinh215 @mvar ??

File: Command/IndexImportCommand.php (hunk 1 above)
Comment (ndinh215) [inline Command/IndexImportCommand.php]: Total agree, @saimaz

File: Service/Json/JsonReader.php (hunk 3 above)
Comment (ndinh215) [inline Service/Json/JsonReader.php]: I think the options here is the options of the command. Thus, ```$convertDocuments``` should not be put into the options.

**Comment by saimaz** [comment]
I'm looking at the JsonReader and I think that maybe much better would be to include in it `ParametersBagTrait`. Then all options has to be set via setters and we won't need so long constructor. Also it's very difficult to maintain constructor, we simply cannot change the structure.

File: Service/Json/JsonReader.php (hunk 3 above)
Comment (saimaz) [inline Service/Json/JsonReader.php]: I took a bit deeper look and don't think we need this argument. Nothing in the whole bundle uses this option. And by current logic nothing will. Propose to remove `$convertDocuments`. And then remove `convertDocument()` function.

Comment (ndinh215) [inline Service/Json/JsonReader.php]: OK, got it, @saimaz . Thanks.

**Comment by ndinh215** [comment]
I removed the function convertDocument and refactored again. Please review it. Thanks.

**Pull request merged**